compose/project.py

```diff
@@ -695,7 +695,9 @@ def push(self, service_names=None, ignore_push_failures=False):
             repo, tag, sep = parse_repository_tag(service.image_name)
             service_image_name = sep.join((repo, tag)) if tag else sep.join((repo, 'latest'))
```

Comment:
Perhaps `service.image_name` should be modified to do this (note: I'm not a Python developer, so perhaps there's reasons why that wasn't done 😅);  https://github.com/docker/compose/blob/440c94ea7a7e62b3de50722120ca34c4e818205a/compose/service.py#L386-L389

Which could look  something like (again, I'm not a python developer, so possibly there's a cleaner way):

```python
def image_name(self):
    image = self.options.get('image', '{project}_{s.name}'.format(
        s=self, project=self.project.lstrip('_-')
    ))
    if image == "":
        return
    repo, tag, separator = parse_repository_tag(image)
    tag = tag or 'latest'
    return repo+separator+tag
```

Reply:
Why do we need to do so? Just to ensure `tag` is not empty?

Reply:
There's some old behavior where `docker push image<no :tag here>` pushes _all tags_ for `image`. In this case though, omitting `:tag` means `:latest`, so I think it would make sense to return that here (so that this logic doesn't have to be repeated everywhere); open to suggestions though

Reply:
If we look through whole code, we can see, that `latest` is added only after `parse_repository_tag` call. So maybe we should add it there?
Anyways, think we need separate issue/discussion for it (despite it can simplify solving current issue)